OpenRA.Mods.Common/Projectiles/Bullet.cs

```diff
@@ -230,21 +232,25 @@ WAngle GetEffectiveFacing()
 			return new WAngle(effective);
 		}
 
-		public void Tick(World world)
+		public virtual void Tick(World world)
 		{
-			anim?.Tick();
+			Animation?.Tick();
 
 			lastPos = pos;
 			pos = WPos.LerpQuadratic(source, target, angle, ticks, length);
 
 			if (ShouldExplode(world))
+			{
+				if (info.ContrailLength > 0)
+					world.AddFrameEndTask(w => w.Add(new ContrailFader(pos, contrail)));
```

Comment:
This won't be able to execute more than once? Otherwise it seems performance intensive.

Reply:
I noticed later this was also already called before, but in a method below.

Reply:
You mean `ContrailFader`? It's added to the `World` (in `Bullet`) only once - here in this `if`.

Reply:
this only gets called once